torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -12,13 +14,36 @@ using worker_id_t = int16_t;
 // A globally unique ID to identify an RpcAgent
 struct WorkerId {
   WorkerId(std::string name, int id)
-      : name_(std::move(name)), id_(id) {
+      : WorkerId(std::move(name), (worker_id_t)id) {
     TORCH_CHECK(id <= std::numeric_limits<worker_id_t>::max(),
         "RPC worker id ", id, " out of bound of int16_t.");
   }
 
   WorkerId(std::string name, worker_id_t id)
-      : name_(std::move(name)), id_(id) {}
+      : name_(std::move(name)), id_(id) {
+    bool validSize = name_.length() < MAX_NAME_LEN && name_.length() > 0;
```

Comment:
Can we also add unit tests which test various different combinations of valid and invalid worker names?

Reply:
Yes, will add.